terraform/hydrocron-iam.tf

```diff
@@ -44,7 +44,19 @@ data "aws_iam_policy_document" "dynamo-write-policy" {
   }
 
 }
+data "aws_iam_policy_document" "lambda-invoke-policy" {
 
+  statement {
+    effect  = "Allow"
+    actions = [
+      "lambda:InvokeFunction"
+    ]
+    resources = [
+      "arn:aws:lambda:${data.aws_region.current.id}:${local.account_id}:*",
```

Comment:
can we plug the load data invoke arn in here specifically? I think we can refer to it with `aws_lambda_function.hydrocron_lambda_load_granule.invoke_arn`

Reply:
@frankinspace I'm having a really hard time getting around Cycle Errors in terraform when I try to specify the arn explicitly. I've tried using variables for the role name in the iam role definition, in both lambda function definitions, and for the invoke arn in the policy doc and can't find a way around the circular dependency. Any suggestions on what else I can try?

Reply:
Oh I see. I think having two roles instead of a single shared one would remove the cycle.

load data -> load data role -> permission to invoke load granule (and s3 read I think?)
load granule -> load granule role -> permission to read s3 and write dynamo

I think that would avoid the circular dependency because load data would depend on load granule but load granule does not depend on load data.

Reply:
Thanks that worked